implementations/rust/ockam/ockam_node/src/storage/database/sqlx_database/migration_support.rs

```diff
@@ -0,0 +1,35 @@
+use crate::database::{FromSqlxError, SqlxDatabase, ToSqlxType, ToVoid};
+use ockam_core::compat::time::now;
+use ockam_core::errcode::{Kind, Origin};
+use ockam_core::Result;
+use sqlx::sqlite::SqliteRow;
+use sqlx::{query, Row, SqlitePool};
+use time::OffsetDateTime;
+
+impl SqlxDatabase {
+    pub(crate) async fn has_migrated(pool: &SqlitePool, migration_name: &str) -> Result<bool> {
+        let query = query("SELECT COUNT(*) FROM _rust_migrations WHERE name=?")
+            .bind(migration_name.to_sql());
+        let count_raw: Option<SqliteRow> = query.fetch_optional(pool).await.into_core()?;
+
+        if let Some(count_raw) = count_raw {
+            let count: i64 = count_raw.get(0);
+            Ok(count != 0)
+        } else {
+            Ok(false)
+        }
+    }
+
+    pub(crate) async fn mark_as_migrated(pool: &SqlitePool, migration_name: &str) -> Result<()> {
+        let now = now()?;
```

Comment:
If we have `now()` returning a `TimestampInSeconds` we can store it directly because it already has a `ToSqlxType` instance.

Reply:
The problem is that `TimestampInSeconds` is in `ockam_identity` crate, and I'm not sure we should move it